gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java

```diff
@@ -63,7 +63,8 @@ public class MysqlSpecStore extends MysqlBaseSpecStore {
   // record may contain data there... though in practice, all should, given the passage of time (amidst the usual retention expiry).
   protected static final String SPECIFIC_INSERT_STATEMENT = "INSERT INTO %s (spec_uri, flow_group, flow_name, template_uri, "
       + "user_to_proxy, source_identifier, destination_identifier, schedule, tag, isRunImmediately, owning_group, spec, spec_json) "
-      + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE spec = VALUES(spec), spec_json = VALUES(spec_json)";
+      + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE template_uri = VALUES(template_uri),user_to_proxy = VALUES(user_to_proxy),source_identifier = VALUES(source_identifier),schedule = VALUES(schedule),tag = VALUES(tag),isRunImmediately = VALUES(isRunImmediately),owning_group = VALUES(owning_group),"
```

Comment:
I wasn't familiar w/ the `ON DUPLICATE KEY` ... `VALUES()` function.  when I looked it up, it appears to be deprecated and row/column aliases are to be used instead.  see: https://dev.mysql.com/doc/refman/8.4/en/insert-on-duplicate.html

looks like `destination_identifier` may have been omitted.  keeping these in the same order as they're listed prior (right after `INSERT INTO %s`) might make that more noticeable

finally, as in java source, leave a space after `,` in SQL

Reply:
thanks for the detail,addressed

Reply:
the alias way is only supported on mysql 8.0.20 and above. Our test suites currently use wix-embedded-mysql and the latest mysql version it has is 8.0.17 [here](https://github.com/wix-incubator/wix-embedded-mysql/blob/master/wix-embedded-mysql/src/main/java/com/wix/mysql/distribution/Version.java)
so the test is failing due to unsupported query, as the row/column aliases way  was introduced in 8.0.20.
so have kept the ON DUPLICATE KEY ... VALUES() way of updating for now, and maybe we can create a different ticket to move away from wix-embedded-mysql and explore other alternatives. How do you suggest we go ahead?

Reply:
Have updated the mysql test version in this [PR](https://github.com/apache/gobblin/pull/3986)